pkg/stream/output_channel.go

```diff
@@ -59,3 +60,7 @@ func (c *outputChannel) Close() {
 		c.logger.Warn("duplicate close to output channel: channel is already closed")
 	}
 }
+
+func (c outputChannel) IsClosed() bool {
+	return c.closed
```

Comment:
you are not allowed to read closed without taking the lock for reading (otherwise you have a race and the race detector might scream)